redfish-core/lib/processor.hpp

```diff
@@ -1160,6 +1194,120 @@ inline void handleProcessorCollectionHead(
+                     const dbus::utility::MapperServiceMap& serviceMap,
+                     const std::string& objectPath,
+                     std::optional<bool> locationIndicatorActive)
+{
+    if (locationIndicatorActive)
+    {
+        for (const auto& [serviceName, interfaceList] : serviceMap)
+        {
+            bool cpuInterface = false;
+            bool associationInterface = false;
+            for (const auto& interface : interfaceList)
+            {
+                if (interface == "xyz.openbmc_project.Inventory.Item.Cpu")
+                {
+                    cpuInterface = true;
+                }
+                else if (interface ==
+                         "xyz.openbmc_project.Association.Definitions")
+                {
+                    associationInterface = true;
+                }
+            }
+
+            if (cpuInterface && associationInterface)
+            {
+                setLocationIndicatorActive(asyncResp, objectPath,
+                                           *locationIndicatorActive);
+            }
+        }
+    }
+}
+
+inline void afterSetProcessorObject(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& processorId, std::optional<bool> locationIndicatorActive,
+    const boost::system::error_code& ec,
+    const dbus::utility::MapperGetSubTreeResponse& subtree)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error {}", ec.value());
+        messages::internalError(asyncResp->res);
+        return;
+    }
+    for (const auto& [objectPath, serviceMap] : subtree)
+    {
+        // Ignore any objects which don't end with our desired cpu name
+        sdbusplus::message::object_path path(objectPath);
+        std::string name = path.filename();
+        if (name.empty() || name != processorId)
+        {
+            continue;
+        }
+
+        bool found = false;
+        // Filter out objects that don't have the CPU-specific
+        // interfaces to make sure we can return 404 on non-CPUs (e.g.
+        // /redfish/../Processors/dimm0)
+        for (const auto& [serviceName, interfaceList] : serviceMap)
+        {
+            if (std::find_first_of(interfaceList.begin(), interfaceList.end(),
+                                   processorInterfaces.begin(),
+                                   processorInterfaces.end()) !=
+                interfaceList.end())
+            {
+                found = true;
+                break;
+            }
+        }
+
+        if (!found)
+        {
+            continue;
+        }
+
+        // Process the first object which does match our cpu name and
+        // required interfaces, and potentially ignore any other
+        // matching objects. Assume all interfaces we want to process
+        // must be on the same object path.
+        setProcessorData(asyncResp, serviceMap, objectPath,
+                         locationIndicatorActive);
```

Comment:
I think L1257-L1279 be can be simplified like this?

```
    for (const auto& [serviceName, interfaceList] : serviceMap)
    {
        if (std::find_first_of(interfaceList, processorInterfaces) !=  interfaceList.end())
        {
            // Process the first object which does match our cpu name and
            // required interfaces, and potentially ignore any other
            // matching objects. Assume all interfaces we want to process
            // must be on the same object path.
            setProcessorData(asyncResp, serviceMap, objectPath,
                                    locationIndicatorActive);
            return;
        }
    }
    messages::resourceNotFound(asyncResp->res, "#Processor.v1_18_0.Processor",
                               processorId);
```
```

Reply:
Good simplification, I've changed it.